tools/migration/rdbms/README.md

```diff
@@ -0,0 +1,99 @@
+* `0001_add_extended_descriptor.sql`
+* `0002_migrate_descriptor_to_extended_descriptor.go`
+
+
+The output of `status` will looks as follows:
+```
+go run tools/migration/rdbms/main.go -dir db/rdbms/migration status 
+2020/12/09 19:02:45     Applied At                  Migration
+2020/12/09 19:02:45     =======================================
+2020/12/09 19:02:45     Pending                  -- 0001_add_extended_descriptor.sql
+2020/12/09 19:02:45     Pending                  -- 0002_migrate_descriptor_to_extended_descriptor.go
+```
+
+We can now proceed applying the migrations.
+
+# Up command
+`up` command, and its variations, can be used run forward migrations. Taking for example the output of the `status` command above, we can run both migrations in sequence:
+
+```
+go run tools/migration/rdbms/main.go -dir db/rdbms/migration up
+2020/12/09 19:11:04 OK    0001_add_extended_descriptor.sql
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering target manager csvfiletargetmanager
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering target manager targetlist
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test fetcher uri
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test fetcher literal
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test step echo
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test step slowecho
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test step example
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test step cmd
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test step sshcmd
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering test step randecho
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering reporter targetsuccess
+[2020-12-09T19:11:04Z]  INFO pkg/pluginregistry: Registering reporter noop
+2020/12/09 19:11:04 OK    0002_migrate_descriptor_to_extended_descriptor.go
+2020/12/09 19:11:04 goose: no migrations to run. current version: 2
+```
+
+The tool indicates that the version of the schema has been brought up to `2`. Output of `status` confirms this:
+```
+ $ go run tools/migration/rdbms/main.go -dir db/rdbms/migration status
+2020/12/09 19:12:05     Applied At                  Migration
+2020/12/09 19:12:05     =======================================
+2020/12/09 19:12:05     Wed Dec  9 19:11:04 2020 -- 0001_add_extended_descriptor.sql
+2020/12/09 19:12:05     Wed Dec  9 19:11:04 2020 -- 0002_migrate_descriptor_to_extended_descriptor.go
+```
+
+# Down command
+`down` is the counterpart of `up`, as it rewinds the migration. Note that what actually happens when a migration is executed on the `down` path is very dependent on the implementation of the migration itself. `down` migrations might be a no-op, in which case the tool will only roll back the version, but no actual work will be executed. 
```

Comment:
Is the no-op different from a down migration not being supported, or is that the same thing?

If they are the same, should we not fail and say "Sry can't go down"?

Reply:
It's the same, and I have been thinking of making it fail hard, but I couldn't find a non-ugly way to do so in `.sql` migrations and during testing it was convenient to be able to reset the db version. But you are right, failing hard is the right thing to do. 

I'll sort out the first internal migration and put up a PR to do so.